src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -42,14 +42,14 @@ public class DeveloperExceptionPageMiddleware
         /// <param name="options"></param>
         /// <param name="loggerFactory"></param>
         /// <param name="hostingEnvironment"></param>
-        /// <param name="diagnosticSource"></param>
+        /// <param name="diagnosticListener"></param>
         /// <param name="filters"></param>
         public DeveloperExceptionPageMiddleware(
             RequestDelegate next,
             IOptions<DeveloperExceptionPageOptions> options,
             ILoggerFactory loggerFactory,
             IWebHostEnvironment hostingEnvironment,
-            DiagnosticSource diagnosticSource,
+            DiagnosticListener diagnosticListener,
```

Comment:
This is a breaking change that can easily be avoided.

Reply:
Means lose the fast inlining `diagnosticListerner.IsEnabled()` test and have to use the slow virtual `diagnosticSource.IsEnabled(string))`

Reply:
Suppose it should be enabled in production